.circleci/config.yml

```diff
@@ -0,0 +1,246 @@
+
+      - run:
+          name: Install local pip packages
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            paver test_system -s cms --with-flaky --cov-args="-p" --with-xunitmp
+
+# ########## LIB UNIT TESTS ##########
+
+  lib_unit_tests:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    environment:
+      - NO_PREREQ_INSTALL: "true"
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Install local pip packages
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            paver test_lib --with-flaky --cov-args="-p" --with-xunitmp
+
+# ########## JAVASCRIPT TESTS ##########
+
+  javascript_tests:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    environment:
+      - NO_PREREQ_INSTALL: "true"
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Install local pip packages
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
```

Comment:
@symbolist why aren't this installed as part of the build section?

Reply:
This installs the libs in `common/lib/` as packages with a `-e`. As far as I understand python packaging, this causes the `egginfo` to be created in `common/lib/<package>/` and `venv/lib/python2.7/site-packages/` contains a `egg-link` to it. Because the source code is being checked out in each build the `egginfo` is missing from `common/lib/<package>`. This sets up everything again.